src/compiler/parser.ts

```diff
@@ -2973,6 +2988,32 @@ namespace ts {
             return parseParenthesizedArrowFunctionExpressionHead(/*allowAmbiguity*/ false);
         }
 
+        function tryParseAsyncSimpleArrowFunctionExpression(): ArrowFunction {
+            const isUnParenthesizedAsyncArrowFunction = lookAhead(isUnParenthesizedAsyncArrowFunctionWorker);

```

Comment:
can you check first that  `token === SyntaxKind.AsyncKeyword`  before calling the function?